client-js/app/scripts/app.ts

```diff
@@ -105,38 +105,28 @@ window.addEventListener('WebComponentsReady', () => {
 
   var activity: Activity = null;
 
-  function activate(newActivity: Activity): void {
-    var p: Promise<void> = Promise.resolve(null);
-    if (activity) {
-      p = activity.stop();
-    }
```

Comment:
That's correct. This kind of conditional logic is tricky when chaining promises because you have to play tricks to ensure that all code branches get chained together properly. async/await methods just let you use all the conventional control structures and all the branches get tied together under the hood.